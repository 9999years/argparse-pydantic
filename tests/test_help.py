import pytest
from pytest import CaptureFixture
from syrupy.assertion import SnapshotAssertion

from pydantic import Field
from pydantic_argparse_extensible import ArgModel


class Args(ArgModel):
    user: str = Field(description="""The username to connect to.""")


def test_help(capsys: CaptureFixture, snapshot: SnapshotAssertion) -> None:
    with pytest.raises(SystemExit):
        Args.argparse(args=["--help"])
    assert capsys.readouterr().out == snapshot
