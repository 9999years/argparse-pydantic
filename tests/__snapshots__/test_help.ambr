# serializer version: 1
# name: test_help
  '''
  usage: pytest [-h] --user USER
  
  options:
    -h, --help   show this help message and exit
    --user USER  The username to connect to.
  
  '''
# ---
